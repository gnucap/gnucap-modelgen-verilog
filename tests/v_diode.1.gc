// simple diode test, d_diode.1.ckt
verilog
attach ../vams/diode.so

verilog

paramset ddd diode;
  parameter real area=1;
 .area= area;
 .level=0;
 .is= 10.f;
 .rs= 0.;
 .n= 1.;
 .tt= 0.;
 .cjo= 1.p;
 .vj= 1.;
 .m= 0.5 ;
 .eg= 1.11;
 .xti= 3.;
 .kf= 0.;
 .af= 1.;
 .fc= 0.5;
 .bv= 50;
 .ibv= 0.001;
endparamset;

module main(1 2)
  ddd #(.area(1.)) d1 (1  2);
endmodule

main m(2 0);

spice

V1   1  0  10. AC 1
R1   1  2  100.K
*>.list
*>.print op v(1) v(2) i(v1) 
*>.print op + i(r1) id(m.d1) vd(m.d1) req(m.d1) cd(m.d1) z(m.d1) zraw(m.d1) gd(m.d1) z(2) iter(0) region(m.d1)
.op
.print dc v(1) v(2) i(v1) 
*>.print dc + i(r1) id(m.d1) vd(m.d1) req(m.d1) cd(m.d1) z(m.d1) zraw(m.d1) gd(m.d1) z(2) iter(0) region(m.d1)
.dc v1 -10 10 2
*>.stat notime
*>.op
*>.stat notime
.end
