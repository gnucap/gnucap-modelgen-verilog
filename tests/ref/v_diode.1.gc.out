.subckt ddd (  )
parameter real area=1; 
.ends ddd
.subckt main ( 1 2 )
d1 ( 1 2 )  ddd  area=1.
.ends main
m ( 2 0 )  main 
V1 ( 1 0 )  DC  10.  AC  1.
R1 ( 1 2 )  100.K
#           v(1)       v(2)       i(V1)      i(R1)      id(m.d1)   vd(m.d1)   req(m.d1)  cd(m.d1)   z(m.d1)    zraw(m.d1) gd(m.d1)   z(2)       iter(0)    region(m.d1)
 27.        10.        0.59396   -94.06u     94.06u     ??         ??         ??         ??         ??         ??         ??         274.22     8.         1.        
#           v(1)       v(2)       i(V1)      i(R1)      id(m.d1)   vd(m.d1)   req(m.d1)  cd(m.d1)   z(m.d1)    zraw(m.d1) gd(m.d1)   z(2)       iter(0)    region(m.d1)
-10.       -10.       -10.        0.        -10.01p     ??         ??         ??         ??         ??         ??         ??         100.K      5.        -1.        
-8.        -8.        -8.         0.        -8.01p      ??         ??         ??         ??         ??         ??         ??         100.K      4.        -1.        
-6.        -6.        -6.         0.        -6.01p      ??         ??         ??         ??         ??         ??         ??         100.K      4.        -1.        
-4.        -4.        -4.         0.        -4.01p      ??         ??         ??         ??         ??         ??         ??         100.K      4.        -1.        
-2.        -2.        -2.         0.        -2.01p      ??         ??         ??         ??         ??         ??         ??         100.K      4.        -1.        
 0.         0.         0.         0.         0.         ??         ??         ??         ??         ??         ??         ??         100.K      5.        -1.        
 2.         2.         0.54568   -14.543u    14.543u    ??         ??         ??         ??         ??         ??         ??         1.7474K    10.        1.        
 4.         4.         0.56789   -34.321u    34.321u    ??         ??         ??         ??         ??         ??         ??         747.96     5.         1.        
 6.         6.         0.5797    -54.203u    54.203u    ??         ??         ??         ??         ??         ??         ??         474.91     4.         1.        
 8.         8.         0.5878    -74.122u    74.122u    ??         ??         ??         ??         ??         ??         ??         347.73     4.         1.        
 10.        10.        0.59396   -94.06u     94.06u     ??         ??         ??         ??         ??         ??         ??         274.22     4.         1.        
Gnucap   System status
iterations: op=8, dc=53, tran=0, fourier=0, total=61
transient timesteps: accepted=0, rejected=0, total=0
nodes: user=2, subckt=0, model=0, total=2
dctran density=100.0%, ac density=100.0%
#           v(1)       v(2)       i(V1)      i(R1)      id(m.d1)   vd(m.d1)   req(m.d1)  cd(m.d1)   z(m.d1)    zraw(m.d1) gd(m.d1)   z(2)       iter(0)    region(m.d1)
 27.        10.        0.59396   -94.06u     94.06u     ??         ??         ??         ??         ??         ??         ??         274.22     8.         1.        
Gnucap   System status
iterations: op=8, dc=53, tran=0, fourier=0, total=69
transient timesteps: accepted=0, rejected=0, total=0
nodes: user=2, subckt=0, model=0, total=2
dctran density=100.0%, ac density=100.0%
